MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -9,240 +15,341 @@ namespace MediaBrowser.Common.Net
-        /// <param name="excludeList">Items to exclude.</param>
-        /// <param name="isNetwork">Collection is a network collection.</param>
-        /// <returns>A new collection, with the items excluded.</returns>
-        public static Collection<IPObject> Exclude(this Collection<IPObject> source, Collection<IPObject> excludeList, bool isNetwork)
+        /// <param name="host">Host name to parse.</param>
+        /// <param name="addresses">Object representing the span, if it has successfully been parsed.</param>
+        /// <param name="isIPv4Enabled"><c>true</c> if IPv4 is enabled.</param>
+        /// <param name="isIPv6Enabled"><c>true</c> if IPv6 is enabled.</param>
+        /// <returns><c>true</c> if the parsing is successful, <c>false</c> if not.</returns>
+        public static bool TryParseHost(ReadOnlySpan<char> host, [NotNullWhen(true)] out IPAddress[] addresses, bool isIPv4Enabled = true, bool isIPv6Enabled = false)
         {
-            if (source.Count == 0 || excludeList is null)
+            if (host.IsEmpty)
             {
-                return new Collection<IPObject>(source);
+                addresses = Array.Empty<IPAddress>();
+                return false;
             }
 
-            Collection<IPObject> results = new Collection<IPObject>();
+            host = host.Trim();
 
-            bool found;
-            foreach (var outer in source)
+            // See if it's an IPv6 with port address e.g. [::1] or [::1]:120.
+            if (host[0] == '[')
             {
-                found = false;
+                int i = host.IndexOf("]", StringComparison.Ordinal);
+                if (i != -1)
+                {
+                    return TryParseHost(host[1..(i - 1)], out addresses);
+                }
+
+                addresses = Array.Empty<IPAddress>();
+                return false;
+            }
 
-                foreach (var inner in excludeList)
+            var hosts = new List<string>();
+            foreach (var splitSpan in host.Split(':'))
+            {
+                hosts.Add(splitSpan.ToString());
+            }
+
+            if (hosts.Count <= 2)
+            {
+                // Is hostname or hostname:port
+                if (_fqdnRegex.IsMatch(hosts[0]))
                 {
-                    if (outer.Equals(inner))
+                    try
                     {
-                        found = true;
-                        break;
+                        addresses = Dns.GetHostAddresses(hosts[0]);
+                        return true;
+                    }
+                    catch (SocketException)
+                    {
+                        // Log and then ignore socket errors, as the result value will just be an empty array.
+                        Console.WriteLine("GetHostAddresses failed.");
                     }
                 }
 
-                if (!found)
+                // Is an IP4 or IP4:port
+                host = hosts[0].Split('/')[0];
```

Comment:
We have an extension called LeftPart or something that returns the span up to the delimiter